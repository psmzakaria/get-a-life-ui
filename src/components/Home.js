import React, { Component } from "react";
import { Grid, Responsive } from "semantic-ui-react";
import SideInfo from "./SideInfo";
import CredentialsForm from "./CredentialsForm";
import { API_URL } from "../utils/configVar";
import Logo from "./Logo";

class Home extends Component {
  constructor() {
    super();
    this.state = {
      componentToDisplay: "Sign Up",
      signUpError: {
        hidden: true,
        header: "",
        content: ""
      },
      signInError: {
        hidden: true,
        header: "",
        content: ""
      },
      username: "",
      password: ""
    };
  }

  render() {
    return (
      <div className="homeBgImage">
        
        <div className="logoPadding">
          <Logo />
        </div>
        <div className="container">
          <Responsive as={Grid} minWidth={768}>
            <Grid.Column width={8}>
              <SideInfo />
            </Grid.Column>
            <Grid.Column className="" width={8}>
              <CredentialsForm
                loadComponent={this.loadComponent}
                handleOnChange={this.handleOnChange}
                handleOnSubmit={this.handleOnSubmit}
                state={this.state}
              />
            </Grid.Column>
          </Responsive>
          <Responsive as={Grid} {...Responsive.onlyMobile}>
            <Grid.Row>
              <Grid.Column>
                <SideInfo />
              </Grid.Column>
            </Grid.Row>
            <Grid.Row>
              <Grid.Column>
                <CredentialsForm
                  loadComponent={this.loadComponent}
                  handleOnChange={this.handleOnChange}
                  handleOnSubmit={this.handleOnSubmit}
                  state={this.state}
                />
              </Grid.Column>
            </Grid.Row>
          </Responsive>
        </div>
      </div>
    );
  }

  handleOnChange = event => {
    this.setState({
      [event.target.id]: event.target.value
    });
  };

  handleOnSubmit = async event => {
    event.preventDefault();

    const response = await fetch(`${API_URL}/account/${event.target.id}`, {
      method: "POST",
      credentials: "include",
      headers: {
        Accept: "application/json",
        "Content-Type": "application/json"
      },
      body: JSON.stringify({
        username: this.state.username,
        password: this.state.password
      })
    });

    const data = await response.json();

    if (response.ok) {
      this.props.history.push({
        pathname: `${this.state.username}/events`
      });
    } else {
      if (this.state.componentToDisplay === "Sign Up") {
        this.setState({
          signUpError: {
            hidden: false,
            header: "Error",
            content: data
          }
        });
      } else {
        this.setState({
          signInError: {
            hidden: false,
            header: "Error",
            content: data.message
          }
        });
      }
    }
  };

  loadComponent = componentName => {
    this.setState({
      componentToDisplay: componentName
    });
  };
}

export default Home;
