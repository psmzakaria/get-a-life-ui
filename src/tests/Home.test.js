import React from "react";
import Home from "./../components/Home";
import ShallowRenderer from "react-test-renderer/shallow";

it("renders the home component of the app ", () => {
  const renderer = new ShallowRenderer();
  renderer.render(<Home />);
  const output = renderer.getRenderOutput();
  expect(output).toMatchSnapshot();
});
