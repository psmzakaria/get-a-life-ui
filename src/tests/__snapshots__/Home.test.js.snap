// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the home component of the app  1`] = `
<div
  className="container"
>
  <Responsive
    as={[Function]}
    getWidth={[Function]}
    minWidth={768}
  >
    <GridColumn
      width={8}
    >
      <h1>
        logo container
      </h1>
    </GridColumn>
    <GridColumn
      width={8}
    >
      <UserSignup
        loadComponent={[Function]}
      />
    </GridColumn>
  </Responsive>
  <Responsive
    as={[Function]}
    getWidth={[Function]}
    maxWidth={767}
    minWidth={320}
  >
    <GridRow>
      <GridColumn>
        <h1>
          logo container
        </h1>
      </GridColumn>
    </GridRow>
    <GridRow>
      <GridColumn>
        <UserSignup />
      </GridColumn>
    </GridRow>
  </Responsive>
</div>
`;
